### ISSUE TITLE
CI broken on master

### ISSUE THREAD

**Issue by astrojuanlu** [poliastro/poliastro#1100]
See https://github.com/poliastro/poliastro/pull/1027, https://github.com/poliastro/poliastro/pull/1098.

- Several `mypy` errors due to the latest NumPy version actually shipping types:

```
2021-02-09T22:43:02.2980955Z =================================== FAILURES ===================================
2021-02-09T22:43:02.2981458Z _________________________________ test session _________________________________
2021-02-09T22:43:02.2981817Z mypy exited with status 1.
2021-02-09T22:43:02.2992268Z ____________________________ src/poliastro/ephem.py ____________________________
2021-02-09T22:43:02.2992738Z 60: error: No overload variant of "linspace" matches argument types "Any", "int"
2021-02-09T22:43:02.2993204Z 60: note: Possible overload variants:
2021-02-09T22:43:02.2999138Z 60: note:     def linspace(start: Union[Union[int, float, complex, number[Any], bool_], Sequence[Union[int, float, complex, number[Any], bool_]], ndarray, _SupportsArray, Sequence[Sequence[Any]]], stop: Union[Union[int, float, complex, number[Any], bool_], Sequence[Union[int, float, complex, number[Any], bool_]], ndarray, _SupportsArray, Sequence[Sequence[Any]]], num: SupportsIndex = ..., endpoint: bool = ..., retstep: Literal[False] = ..., dtype: Union[dtype[Any], None, type, _SupportsDType, str, Union[Tuple[Any, int], Tuple[Any, Union[int, Sequence[int]]], List[Any], _DTypeDict, Tuple[Any, Any]]] = ..., axis: SupportsIndex = ...) -> ndarray
2021-02-09T22:43:02.3003008Z 60: note:     def linspace(start: Union[Union[int, float, complex, number[Any], bool_], Sequence[Union[int, float, complex, number[Any], bool_]], ndarray, _SupportsArray, Sequence[Sequence[Any]]], stop: Union[Union[int, float, complex, number[Any], bool_], Sequence[Union[int, float, complex, number[Any], bool_]], ndarray, _SupportsArray, Sequence[Sequence[Any]]], num: SupportsIndex = ..., endpoint: bool = ..., retstep: Literal[True] = ..., dtype: Union[dtype[Any], None, type, _SupportsDType, str, Union[Tuple[Any, int], Tuple[Any, Union[int, Sequence[int]]], List[Any], _DTypeDict, Tuple[Any, Any]]] = ..., axis: SupportsIndex = ...) -> Tuple[ndarray, inexact[Any]]
2021-02-09T22:43:02.3006497Z ________________________ src/poliastro/core/elements.py ________________________
2021-02-09T22:43:02.3007349Z 8: error: Module 'numpy.core.umath' has no attribute 'cos'
2021-02-09T22:43:02.3007926Z 8: error: Module 'numpy.core.umath' has no attribute 'sin'
2021-02-09T22:43:02.3008553Z 8: error: Module 'numpy.core.umath' has no attribute 'sqrt'
2021-02-09T22:43:02.3008974Z ______________________ src/poliastro/czml/extract_czml.py ______________________
2021-02-09T22:43:02.3009512Z 550: error: Module has no attribute "float"; maybe "float_", "cfloat", or "float64"?
2021-02-09T22:43:02.3013421Z ________________________ src/poliastro/twobody/orbit.py ________________________
2021-02-09T22:43:02.3014132Z 1370: error: No overload variant of "linspace" matches argument types "Any", "Any"
2021-02-09T22:43:02.3014545Z 1370: note: Possible overload variants:
2021-02-09T22:43:02.3017068Z 1370: note:     def linspace(start: Union[Union[int, float, complex, number[Any], bool_], Sequence[Union[int, float, complex, number[Any], bool_]], ndarray, _SupportsArray, Sequence[Sequence[Any]]], stop: Union[Union[int, float, complex, number[Any], bool_], Sequence[Union[int, float, complex, number[Any], bool_]], ndarray, _SupportsArray, Sequence[Sequence[Any]]], num: SupportsIndex = ..., endpoint: bool = ..., retstep: Literal[False] = ..., dtype: Union[dtype[Any], None, type, _SupportsDType, str, Union[Tuple[Any, int], Tuple[Any, Union[int, Sequence[int]]], List[Any], _DTypeDict, Tuple[Any, Any]]] = ..., axis: SupportsIndex = ...) -> ndarray
2021-02-09T22:43:02.3034390Z 1370: note:     def linspace(start: Union[Union[int, float, complex, number[Any], bool_], Sequence[Union[int, float, complex, number[Any], bool_]], ndarray, _SupportsArray, Sequence[Sequence[Any]]], stop: Union[Union[int, float, complex, number[Any], bool_], Sequence[Union[int, float, complex, number[Any], bool_]], ndarray, _SupportsArray, Sequence[Sequence[Any]]], num: SupportsIndex = ..., endpoint: bool = ..., retstep: Literal[True] = ..., dtype: Union[dtype[Any], None, type, _SupportsDType, str, Union[Tuple[Any, int], Tuple[Any, Union[int, Sequence[int]]], List[Any], _DTypeDict, Tuple[Any, Any]]] = ..., axis: SupportsIndex = ...) -> Tuple[ndarray, inexact[Any]]
```

- A weird `build` error:

```
2021-02-09T22:42:25.5782954Z check run-test: commands[3] | python -m build .
2021-02-09T22:42:28.6022625Z Found existing installation: setuptools 49.2.1
2021-02-09T22:42:28.6436801Z Uninstalling setuptools-49.2.1:
2021-02-09T22:42:28.6526758Z   Successfully uninstalled setuptools-49.2.1
...
2021-02-09T22:42:35.0995738Z Collecting astroquery>=0.3.9
2021-02-09T22:42:35.1074040Z   Downloading astroquery-0.4.1.tar.gz (6.5 MB)
2021-02-09T22:42:35.6701112Z     ERROR: Command errored out with exit status 1:
2021-02-09T22:42:35.6703383Z      command: /tmp/build-env-260ocgnr/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-gcgcam16/astroquery/setup.py'"'"'; __file__='"'"'/tmp/pip-install-gcgcam16/astroquery/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-3h_ptp3t
2021-02-09T22:42:35.6705974Z          cwd: /tmp/pip-install-gcgcam16/astroquery/
2021-02-09T22:42:35.6706521Z     Complete output (3 lines):
2021-02-09T22:42:35.6707238Z     Traceback (most recent call last):
2021-02-09T22:42:35.6707664Z       File "<string>", line 1, in <module>
2021-02-09T22:42:35.6708402Z     ModuleNotFoundError: No module named 'setuptools'
```

**Comment by DhruvSondhi** [comment]
Hello, According to my understanding ... 
- The latter issue is related to this command being run & removing the `setuptools` library 
```bash
Found existing installation: setuptools 49.2.1
Uninstalling setuptools-49.2.1:
  Successfully uninstalled setuptools-49.2.1
```
I tried reproducing the errors locally but it doesn't seem to be the case :( 
![image](https://user-images.githubusercontent.com/66117751/107461116-4fdf8780-6b7f-11eb-83c1-5e9d88898641.png)

- For the former bug, Found out that the version being installed is `numpy==1.20.1` so this may be the case of deprecated methods & functions usage in `numpy 1.20` as compared to `numpy 1.19`

**Comment by Yash-10** [comment]
For the `mypy` error, there are issues with `numpy==1.20.*` but runs successfully with `numpy==1.19.5`, for instance. 

For the `build` error, it seems that 
```
2021-02-09T22:42:28.6022625Z Found existing installation: setuptools 49.2.1
2021-02-09T22:42:28.6436801Z Uninstalling setuptools-49.2.1:
2021-02-09T22:42:28.6526758Z   Successfully uninstalled setuptools-49.2.1
```
does uninstall already existing `setuptools` but doesn't install it after uninstalling, isn't it?

**Comment by DhruvSondhi** [comment]
Check my previous comment 🙄

**Comment by astrojuanlu** [comment]
The build failure seems to be an upstream bug: https://github.com/pypa/build/issues/231

The NumPy thing should be easy to fix I think, making some adjustments to our code or adding `# type: ignore` where necessary. But I am not sure if other things can appear.

**Comment by Yash-10** [comment]
Oh I see, as you have suggested, the `numpy` errors do get resolved by either using `# type: ignore` or by making changes to the code whenever necessary. But yes, it seems that two additional deprecation warnings pop up when using the recent numpy versions (`1.20.*`) but those deprecations also seem to be concerned with upstreams.

**Comment by astrojuanlu** [comment]
This was partly addressed in #1101

**Comment by astrojuanlu** [comment]
And the second part was fixed in #1102
